addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -283,6 +283,68 @@ def test_on_next_order_reward_promotion_program_with_requirements(self):
         order_bis._update_programs_and_rewards()
         self.assertEqual(len(order_bis.order_line), 2, "Free product from a coupon generated from a promotion program on next order should not dissapear")
 
+    def test_partner_assigned_to_next_order_coupon(self):
+        # Step 1: Create a loyalty program of type 'next_order_coupons'
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': '10% Discount on Next Order',
+            'program_type': 'next_order_coupons',
+            'applies_on': 'future',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {})],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 10,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+
+        # Step 2: Create a sale order
+        order = self.empty_order.copy()
+        self.product_A = self.env['product.product'].create({
+            'name': 'Product A',
+            'list_price': 100,
+            'sale_ok': True
+        })
+
+        order.write({'order_line': [
+            (0, False, {
+                'product_id': self.product_A.id,
+                'name': '1 Product A',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            })
+        ]})
+
+        # Step 3: Apply the loyalty program to the order
```

Comment:
You can make doc sting at start of function about steps it'll look better that way
This is also fine you can keep it if you prefer ;)

Reply:
I think the doc string at start of the function would look better :+1: